spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,136 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe PromptsController do
   include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+  let(:open_signup) do
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
 
   before do
     fake_login
   end
 
-  describe 'no_prompt' do
-    it 'should show an error and redirect' do
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to collection_path(collection)
+      expect(flash[:error]).to eq "What challenge did you want to sign up for?"
+    end
+  end
+
+  describe "no_signup" do
+    it "should show an error and redirect" do
+      signups = create(:challenge_signup)
+      post :create, collection_id: signups.collection.name
+      it_redirects_to collection_signups_path(signups.collection) + "/new"
+      expect(flash[:error]).to eq "Please submit a basic sign-up with the required fields first."
+    end
+  end
+
+  describe "signups_closed" do
+    it "should show an error and redirect" do
+      signups = create(:challenge_signup)
```

Comment:
You use this in a lot of places -- you should put `let(:signup) { create(:challenge_signup) }` at the top, too. (It should be changed to the singular because we're only working with one sign up, not multiple.)